src/System.Memory/src/System/Span.cs

```diff
@@ -0,0 +1,400 @@
+
+        /// <summary>
+        /// Forms a slice out of the given span, beginning at 'start'.
+        /// </summary>
+        /// <param name="start">The index at which to begin this slice.</param>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> index is not in range (&lt;0 or &gt;=Length).
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Span<T> Slice(int start)
+        {
+            if ((uint)start > (uint)_length)
+                throw ThrowHelper.CreateThrowArgumentOutOfRangeException(ExceptionArgument.start);
+
+            IntPtr newOffset = _byteOffset.Add<T>(start);
+            int length = _length - start;
+            return new Span<T>(_pinnable, newOffset, length);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the given span, beginning at 'start', of given length
+        /// </summary>
+        /// <param name="start">The index at which to begin this slice.</param>
+        /// <param name="length">The desired length for the slice (exclusive).</param>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="start"/> or end index is not in range (&lt;0 or &gt;=Length).
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Span<T> Slice(int start, int length)
+        {
+            if ((uint)start > (uint)_length || (uint)length > (uint)(_length - start))
+                throw ThrowHelper.CreateThrowArgumentOutOfRangeException(ExceptionArgument.start);
+
+            IntPtr newOffset = _byteOffset.Add<T>(start);
+            return new Span<T>(_pinnable, newOffset, length);
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into a new array.  This heap
+        /// allocates, so should generally be avoided, however it is sometimes
+        /// necessary to bridge the gap with APIs written in terms of arrays.
+        /// </summary>
+        public T[] ToArray()
+        {
+            if (_length == 0)
+                return SpanHelpers.PerTypeValues<T>.EmptyArray;
+
+            T[] result = new T[_length];
+            CopyTo(result);
+            return result;
+        }
+
+        /// <summary>
+        /// Returns a 0-length span whose base is the null pointer.
+        /// </summary>
+        public static readonly Span<T> Empty = default(Span<T>);
+
+        /// <summary>
+        /// Returns a reference to the 0th element of the Span. If the Span is empty, returns a reference to the location where the 0th element
+        /// would have been stored. Such a reference can be used for pinning but must never be dereferenced.
+        /// </summary>
+        public ref T DangerousGetPinnableReference
```

Comment:
Umm... because that name is long and those parenthesis disappeared from the conscious...